library/core/src/ptr/non_null.rs

```diff
@@ -663,6 +670,12 @@ impl<T: ?Sized> NonNull<T> {
     #[rustc_allow_const_fn_unstable(set_ptr_value)]
     #[stable(feature = "non_null_convenience", since = "1.80.0")]
     #[rustc_const_stable(feature = "non_null_convenience", since = "1.80.0")]
+    //Requires that the input pointer is not null
+    #[kani::requires(!self.as_ptr().is_null())]
```

Comment:
`!self.as_ptr().is_null()` is not necessary because NonNull guarentees no null pointer. Please consider other properties.